typings/index.d.ts

```diff
@@ -3790,15 +3790,24 @@ export interface MessageActivity {
 
 export type MessageAdditions = MessageEmbed | MessageAttachment | (MessageEmbed | MessageAttachment)[];
 
-export interface MessageButtonOptions extends BaseMessageComponentOptions {
-  customId?: string;
+export interface BaseButtonOptions extends BaseMessageComponentOptions {
   disabled?: boolean;
   emoji?: EmojiIdentifierResolvable;
   label?: string;
-  style: MessageButtonStyleResolvable;
-  url?: string;
 }
 
+export type MessageButtonOptions = BaseButtonOptions &
+  (
+    | {
+        style: Exclude<MessageButtonStyle, 'LINK'>;
+        customId: string;
+      }
+    | {
+        style: 'LINK';
+        url: string;
```

Comment:
```suggestion
        style: Exclude<MessageButtonStyleResolvable, 'LINK' | MessageButtonStyles.LINK>;
        customId: string;
      }
    | {
        style: 'LINK' | MessageButtonStyles.LINK;
        url: string;
```

Reply:
Ah good catch!

Reply:
Actually on second thought this may not be needed, `MessageButtonStyleResolvable` isn't an exported type. I'm guessing the only area this would be useful would be for internal implementations.